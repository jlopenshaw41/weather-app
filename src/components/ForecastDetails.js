import React from "react";
import PropTypes from "prop-types";
import WeatherIcon from "react-icons-weather";
import moment from "moment";
import "../styles/ForecastDetails.css";

const ForecastDetails = ({ forecast }) => {
  const { date, temperature, wind, humidity, icon, description } = forecast;
  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">
        {moment(date).format("ddd Do MMM")}
      </div>
      <div className="forecast-details__body">
        <div className="forecast-details__icon">
          <WeatherIcon name="owm" iconId={icon} />
        </div>
        <div className="forecast-details__body__text">
          <div className="forecast-details__description" id="description">
            {description}
          </div>
          <div className="forecast-details__min-temperature">
            Min temp:
            {` ${temperature.min}`}
            &deg;C
          </div>
          <div className="forecast-details__max-temperature">
            Max temp:
            {` ${temperature.max}`}
            &deg;C
          </div>
          <div className="forecast-details__humidity">
            Humidity:
            {` ${humidity}`}
            &#37;
          </div>
          <div className="forecast-details__wind-speed">
            Wind speed:
            {` ${wind.speed}`}
            mph
          </div>
          <div className="forecast-details__wind-direction">
            Wind direction:
            {` ${wind.direction.toUpperCase()}`}
          </div>
        </div>
      </div>
    </div>
  );
};

export default ForecastDetails;

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    description: PropTypes.string.isRequired,
    date: PropTypes.number.isRequired,
    humidity: PropTypes.number.isRequired,
    icon: PropTypes.string.isRequired,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }).isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
  }).isRequired,
};
