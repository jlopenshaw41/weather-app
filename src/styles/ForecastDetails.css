.forecast-details {
    width: 90%;
    padding: 10px 20px;
    align-self: flex-start;
    margin-top: 0px;
}

.forecast-details__date {
  border: 1px red solid;
  align-self: flex-start;  
}