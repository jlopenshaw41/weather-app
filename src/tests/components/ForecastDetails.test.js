import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 11111111,
    humidity: 80,
    temperature: {
      min: 12,
      max: 22,
    },
    wind: {
      speed: 60,
      direction: "ne",
    },
  };

  it("renders as expected", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
