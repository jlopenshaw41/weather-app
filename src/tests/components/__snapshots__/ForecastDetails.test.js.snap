// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders as expected 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu 1st Jan
    </div>
    <div
      class="forecast-details__body"
    >
      <div
        class="forecast-details__icon"
      >
        <i
          class="wi wi-day-sunny"
          name="owm"
        />
      </div>
      <div
        class="forecast-details__body__text"
      >
        <div
          class="forecast-details__description"
          id="description"
        >
          Clear
        </div>
        <div
          class="forecast-details__min-temperature"
        >
          Min temp: 12°C
        </div>
        <div
          class="forecast-details__max-temperature"
        >
          Max temp: 22°C
        </div>
        <div
          class="forecast-details__humidity"
        >
          Humidity: 80%
        </div>
        <div
          class="forecast-details__wind-speed"
        >
          Wind speed: 60mph
        </div>
        <div
          class="forecast-details__wind-direction"
        >
          Wind direction: NE
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
